### ISSUE TITLE
When using voiceover to navigate the document page, the focus on the links is not logical in iOS.

### ISSUE THREAD

**Issue by msftedad** [mozilla/pdf.js#19169]
### Attach (recommended) or Link to PDF file

https://mozilla.github.io/pdf.js/web/viewer.html

### Web browser and its version

Safari Version 18.1.1

### Operating system and its version

iPhone 13 Version 18.1.1

### PDF.js version

1.4

### Is the bug present in the latest PDF.js version?

Yes

### Is a browser extension

Yes

### Steps to reproduce the problem

1. Open this PDF viewer https://mozilla.github.io/pdf.js/web/viewer.html.
2. Use swipe gestures to move to the 'Tools' option at header at header of pdf viewer.
3. Activate the 'Open' control using double tap and upload any pdf (Having links in it (e.g.: Responsible Computing User Agreement 5-4-2022 (1) (1).pdf)) from your system.
4. Now you'll have the pdf file, in pdf viewer. 
5. Verify using swipe gestures whether the focus on the links is logical or not.

### What is the expected behavior?

Voiceover focus should be logical and visible on the links when user navigates using swipe gestures and must announce the complete link information on the document page in iOS.

### What went wrong?

When using voiceover to navigate the document page, the focus on the links is not logical in iOS.
When using voice over to navigate the links, the focus order is not visible on the selected link. Instead, it moves to the beginning of the document page and announces it as a link.

https://github.com/user-attachments/assets/650def1d-c481-414e-9668-143513314de6
